src/libraries/System.Net.Ping/ref/System.Net.Ping.cs

```diff
@@ -62,10 +62,12 @@ public partial class Ping : System.ComponentModel.Component
         public System.Threading.Tasks.Task<System.Net.NetworkInformation.PingReply> SendPingAsync(System.Net.IPAddress address, int timeout) { throw null; }
         public System.Threading.Tasks.Task<System.Net.NetworkInformation.PingReply> SendPingAsync(System.Net.IPAddress address, int timeout, byte[] buffer) { throw null; }
         public System.Threading.Tasks.Task<System.Net.NetworkInformation.PingReply> SendPingAsync(System.Net.IPAddress address, int timeout, byte[] buffer, System.Net.NetworkInformation.PingOptions? options) { throw null; }
+        public System.Threading.Tasks.Task<System.Net.NetworkInformation.PingReply> SendPingAsync(System.Net.IPAddress address, System.TimeSpan timeout, byte[]? buffer = null, System.Net.NetworkInformation.PingOptions? options = null, System.Threading.CancellationToken cancellationToken = default) { throw null; }
```

Comment:
As mentioned [here](https://github.com/dotnet/runtime/issues/67260#issuecomment-1185972901), I think this API would be more useful if `timeout` were an optional parameter. Requiring it here when other overloads allow you to skip it will likely lead to either poorly-chosen timeouts or copying the default value.

Reply:
@rzikm thoughts on this change?

Reply:
I would stick to what was approved during the API review, any changes to that need to be green-lighted by the API review group.

Reply:
@rzikm I'm fine to keep going with the approved API if that's the direction you want to take.  Just out of curiosity 

* What is the process for getting such a change re-reviewed?
* API review group aside, what is your personal view on which version is better? In my view switching to `TimeSpan? timeout = null` is quite a large improvement because as a caller I wouldn't know what a "reasonable" default timeout was. Interested to hear other opinions, though.

Reply:
_Many_ APIs throughout the core libraries accept a timeout, none to my knowledge accept a nullable TimeSpan, and we wouldn't diverge for this one method. There's not really a reasonable default this method could pick other than infinite or system-default.

Reply:
> as a caller I wouldn't know what a "reasonable" default timeout was

A user has a much better chance of picking an appropriate timeout for their use case than we do.

Reply:
Makes sense thanks @stephentoub , especially if this would be a new pattern. I'll note that other existing Ping overloads *do* use a default timeout (5s), but it sounds like we're moving away from that rather than trying to match it.